cpp/src/writer/tsfile_writer.h

```diff
@@ -61,13 +65,26 @@ class TsFileWriter {
     int register_aligned_timeseries(
         const std::string &device_id,
         const std::vector<MeasurementSchema *> &measurement_schemas);
+    void register_table(const std::shared_ptr<TableSchema>& table_schema);
     int write_record(const TsRecord &record);
     int write_tablet(const Tablet &tablet);
     int write_record_aligned(const TsRecord &record);
     int write_tablet_aligned(const Tablet &tablet);
-    std::map<std::string, MeasurementSchemaGroup *> *get_schema_group_map() {
-        return &schemas_;
-    }
+    int write_table(const Tablet &tablet);
```

Comment:
Would be better to create a new class like TsFileTabletWriter wrapping this class, and only public the Table model interfaces in it.

Reply:
Fixed

class TsFileTableWriter {
   public:
    TsFileTableWriter(WriteFile* writer_file,
                      const std::shared_ptr<TableSchema>& table_schema,
                      uint64_t memory_threshold = 0);
    ~TsFileTableWriter();
    int write_table(const Tablet& tablet);
    int flush();
    int close();
   private:
    std::shared_ptr<TsFileWriter> tsfile_writer_;
};